sdk/identity/azure-identity/azure/identity/aio/_authn_client.py

```diff
@@ -48,6 +49,6 @@ def create_config(**kwargs: Mapping[str, Any]) -> Configuration:
         config = Configuration(**kwargs)
         config.logging_policy = NetworkTraceLoggingPolicy(**kwargs)
         config.retry_policy = AsyncRetryPolicy(
-            retry_on_status_codes=[404, 429] + [x for x in range(500, 600)], **kwargs
+            retry_on_status_codes=[404, 429] + list(range(500, 600)), **kwargs
```

Comment:
>404 [](start = 35, length = 3)

Why are you retrying on 404?

Reply:
Right? According to the [IMDS docs](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#error-handling) the endpoint returns 404 when it's updating, and clients should retry.

Reply:
I'm not a big fan of having to create a list with >100 items. There are some things we can do to improve. Just off the top of my head, something like:

```python
class StatusCodeRange(object):

    def __init__(self, min_value, max_value):
        self.min_value = min_value
        self.max_value = max_value

    def __eq__(self, other):
        return self.min_value <= other and other <= self.max_value


scr = StatusCodeRange(500, 600)

assert 503 in [404, 429, scr]
assert 499 not in [1, 2, scr]
```
It does bastardize the __eq__ method, though, so another version would be to change the simple `in` check in the retry policy to understand ranges as well...

Reply:
> Right? According to the [IMDS docs](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#error-handling) the endpoint returns 404 when it's updating, and clients should retry.

Per the linked documentation (great find, btw :)), we should make sure our retry policy follows the suggested timeouts by default. We are trying to make sure that our libraries "do the right thing" (tm).

Reply:
Have done. I put #5628 on the backlog to track retry code range support.